typespec-tests/src/test/java/com/payload/jsonmergepatch/JsonMergePatchClientTest.java

```diff
@@ -34,41 +51,36 @@ void createAndUpdateResource() {
         resource.setFloatValue(1.1);
         resource.setInnerModel(innerModel);
         resource.setIntArray(Arrays.asList(1, 2, 3));
-        // create resource
-        client.createResource(resource);
-        // update resource
-        ResourcePatch resourcePatch = new ResourcePatch();
-        resourcePatch.setDescription(null);
-        resourcePatch.setArray(null);
-        innerModel.setDescription(null);
-        map.put("key2", null);
-        resourcePatch.setMap(map);
-        resourcePatch.setIntValue(null);
-        resourcePatch.setFloatValue(null);
-        resourcePatch.setInnerModel(null);
-        resourcePatch.setIntArray(null);
-        client.updateResource(resourcePatch);
+        return resource;
     }
 
-    @Test
-    void createAndUpdateOptionalResource() {
+    private static ResourcePatch createResourcePatch() {
         ResourcePatch resourcePatch = new ResourcePatch();
-        resourcePatch.setDescription(null);
-        resourcePatch.setArray(null);
+        resourcePatch.setDescription("desc");
         InnerModel innerModel = new InnerModel();
         innerModel.setName("InnerMadge");
         innerModel.setDescription("innerDesc");
-        innerModel.setDescription(null);
         Map<String, InnerModel> map = new HashMap<>();
         map.put("key", innerModel);
-        map.put("key2", null);
+        List<InnerModel> array = Arrays.asList(innerModel);
+        resourcePatch.setArray(array);
         resourcePatch.setMap(map);
+        resourcePatch.setIntValue(1);
+        resourcePatch.setFloatValue(1.1);
+        resourcePatch.setInnerModel(innerModel);
+        resourcePatch.setIntArray(Arrays.asList(1, 2, 3));
+        return resourcePatch;
+    }
+
+    private static void updateResourcePatch(ResourcePatch resourcePatch) {
```

Comment:
I mean, create this `ResourcePatch` here in this method `Resource Patch updateResourcePatch()`, for the PATCH op. Do not reuse what is from create.

Dev doing JSON Merge Patch do not need to have a resource from create or get.